src/Styleguide/Components/CountdownTimer.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  color,
+  Flex,
+  Sans,
+  space,
+  Spacer,
+  StackableBorderBox,
+} from "@artsy/palette"
+import { WithCurrentTime } from "Components/WithCurrentTime"
+import moment from "moment-timezone"
+import React from "react"
+import styled from "styled-components"
+import { TimerIcon } from "Styleguide/Elements/icons/TimerIcon"
+
+const FIVE_HOURS_IN_SECONDS = 60 * 60 * 5
+
+const pad = (n: number) => n.toString().padStart(2, "0")
+
+const TimeRemaining: React.SFC<{
+  currentTime: string
+  countdownEnd: string
+  highlight: Parameters<typeof color>[0]
+}> = ({ currentTime, countdownEnd, highlight = "purple100" }) => {
+  const timeRemaining = moment.duration(
+    moment(countdownEnd).diff(moment(currentTime))
+  )
+  return (
+    <Sans size="3" color={highlight} weight="medium">
+      {Math.floor(timeRemaining.asSeconds()) <= 0 ? (
+        "0 days"
+      ) : (
+        <>
+          {timeRemaining.days() > 0 && pad(timeRemaining.days()) + "d "}
+          {timeRemaining.hours() > 0 && pad(timeRemaining.hours()) + "h "}
+          {timeRemaining.minutes() > 0 && pad(timeRemaining.minutes()) + "m "}
+          {pad(timeRemaining.seconds()) + "s"}
+        </>
+      )}
+      <span> left</span>
+    </Sans>
+  )
+}
+
+const ProgressBarBackground = styled.div`
+  height: 3px;
+  background-color: ${color("black10")};
+  margin: ${space(0.5)}px 0 ${space(1)}px;
+  display: flex;
+  justify-content: flex-start;
+`
+
+const ProgressBar: React.SFC<{
+  currentTime: string
+  countdownStart: string
+  countdownEnd: string
+  highlight: Parameters<typeof color>[0]
```

Comment:
@ds300 - could you explain this TS pattern a bit more?

Reply:
Sure!

`Parameters<F>` extracts the parameter types from a function type. So, e.g.

```ts
type AbsParams = Parameters<typeof Math.abs>
// => [number]

type CreateElementParams = Parameters<typeof document.createElement>
// => [string, ElementCreationOptions?]

type ConsoleLogParams = Parameters<typeof console.log>
// => [any?, ...any[]]
// (not sure why that isn't just [...any[]])
```

The returned type is always a tuple type, which you can index into to extract the nested elements. So 

    highlight: Parameters<typeof color>[0]

is saying that `highlight` should have the same type as whatever the first argument to the `color` function should have.
 
Now that you brought this up I noticed that there is a `Color` type in palette, which should definitely replace this. I didn't notice it before because I only looked at the type definition for the function `color` which doesn't reference `Color` for some reason.

Reply:
Thank you! this makes perfect sense now.